api/test/e2e/consumer_test.go

```diff
@@ -0,0 +1,202 @@
+ * limitations under the License.
+ */
+package e2e
+
+import (
+	"net/http"
+	"testing"
+	"time"
+)
+
+func TestConsumer_Create_And_Get(t *testing.T) {
+	tests := []HttpTestCase{
+		{
+			caseDesc:     "check consumer is not exist",
+			Object:       MangerApiExpect(t),
+			Path:         "/apisix/admin/consumers/consumer_1",
+			Method:       http.MethodGet,
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusNotFound,
+			ExpectBody:   "",
+		},
+		{
+			caseDesc:     "check consumer is not exist",
+			Object:       MangerApiExpect(t),
+			Path:         "/apisix/admin/consumers/consumer_2",
+			Method:       http.MethodGet,
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusNotFound,
+			ExpectBody:   "",
+		},
+		{
+			caseDesc: "create consumer by POST",
+			Object:   MangerApiExpect(t),
+			Path:     "/apisix/admin/consumers",
+			Method:   http.MethodPost,
+			Body: `{
+				"username": "consumer_1",
+				"plugins": {
+					"limit-count": {
+					"count": 2,
+					"time_window": 60,
+					"rejected_code": 503,
+					"key": "remote_addr"
+					}
+				},
+				"desc": "test description"
+			}`,
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusNotFound,
+			ExpectBody:   "404 page not found",
+		},
+		{
+			caseDesc: "create consumer by PUT",
+			Object:   MangerApiExpect(t),
+			Path:     "/apisix/admin/consumers",
+			Method:   http.MethodPut,
+			Body: `{
+				"username": "consumer_2",
+				"plugins": {
+					"limit-count": {
+					"count": 2,
+					"time_window": 60,
+					"rejected_code": 503,
+					"key": "remote_addr"
+					}
+				},
+				"desc": "test description"
+			}`,
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusOK,
+			ExpectBody:   "",
+			Sleep:        2 * time.Second,
+		},
+		{
+			caseDesc:     "get consumer",
+			Object:       MangerApiExpect(t),
+			Path:         "/apisix/admin/consumers/consumer_2",
+			Method:       http.MethodGet,
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusOK,
+			ExpectBody:   "\"username\":\"consumer_2\"",
+		},
+		{
+			caseDesc: "create consumer without username",
+			Object:   MangerApiExpect(t),
+			Path:     "/apisix/admin/consumers",
+			Method:   http.MethodPut,
+			Body: `{
+				"plugins": {
+					"limit-count": {
+					"count": 2,
+					"time_window": 60,
+					"rejected_code": 503,
+					"key": "remote_addr"
+					}
+				},
+				"desc": "test description"
+			}`,
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusBadRequest,
+			ExpectBody:   "",
+		},
+		{
+			caseDesc:     "delete consumer",
+			Object:       MangerApiExpect(t),
+			Path:         "/apisix/admin/consumers/consumer_2",
+			Method:       http.MethodDelete,
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusOK,
```

Comment:
How to check the delete is successful?

Reply:
The response code is not enough, we need to check in data plane

Reply:
OK, Fixed.